api/src/test/java/org/openmrs/ConceptTest.java

```diff
@@ -570,6 +581,65 @@ public void getSetMembers_shouldReturnUnmodifiableListOfConceptMemberList() {
 		Assert.assertEquals(1, setMembers.size());
 		setMembers.add(new Concept());
 	}
+
+	/**
+	 * @see Concept#getSetMembers()
+	 */
+	@Test
+	public void getSetMembers_shouldReturnTheSameAsgetSetMembersIfincludeRetiredIsTrue() throws Exception {
+		executeDataSet(CONCEPT_XML_DATASET_PACKAGE_PATH);
+
+		Concept c = new Concept();
+
+		Concept setMember1 = service.getConcept(867543);
+		c.addSetMember(setMember1);
+
+		Concept setMember2 = service.getConcept(1234567);
+		c.addSetMember(setMember2);
+
+		Concept setMember3 = service.getConcept(8675439);
+		c.addSetMember(setMember3);
+
+		Concept setMember4 = service.getConcept(12345679);
+		c.addSetMember(setMember4);
+
+
+		List<Concept> setMembers = c.getSetMembers(true);
+
+		Assert.assertEquals(4, setMembers.size());
```

Comment:
Can you confirm that those four Concepts are a mix of retired and unretired Objects? 
You should prove that this setMembers contain the unretired concepts.

Reply:
Here, what I am saying is you are trying to verify your implementation. So you should prove this before using another method. 
i.e : let's count the retired and unretired concepts from setMembers and prove, it contains both.

Reply:
for the retired concepts: Assert.assertEquals(2, c.getSetMembers(false).size());
for the unretired concepts:  Assert.assertEquals(2, c.getSetMembers(true).size() - 
                                               c.getSetMembers(false).size());